crates/meroctl/src/cli/config.rs

```diff
@@ -0,0 +1,172 @@
+    #[arg(long, value_name = "PORT")]
+    pub server_port: Option<u16>,
+
+    /// Enable mDNS discovery
+    #[arg(long, conflicts_with("no_mdns"))]
+    pub mdns: Option<bool>,
+
+    #[arg(long, hide = true, conflicts_with("mdns"))]
+    pub no_mdns: Option<bool>,
+
+    /// Print the config file
+    #[arg(long, short)]
+    pub print: bool,
+}
+
+#[derive(Clone, Debug, ValueEnum)]
+pub enum BootstrapNetwork {
+    CalimeroDev,
+    Ipfs,
+}
+
+impl ConfigCommand {
+    pub fn run(self, root_args: cli::RootArgs) -> eyre::Result<()> {
+        let path = root_args
+            .home
+            .join(&root_args.node_name)
+            .join("config.toml");
+
+        // Load the existing TOML file
+        let toml_str = fs::read_to_string(&path)
+            .map_err(|_| eyre::eyre!("Node must be initialized first."))?;
+        let mut doc = toml_str.parse::<DocumentMut>()?;
+
+        if self.print {
+            println!("{}", doc.to_string());
+            return Ok(());
+        }
+
+        let (ipv4_host, ipv6_host) =
+            self.swarm_host
+                .iter()
+                .fold((None, None), |(v4, v6), ip| match ip {
+                    IpAddr::V4(v4_addr) => (Some(*v4_addr), v6),
+                    IpAddr::V6(v6_addr) => (v4, Some(*v6_addr)),
+                });
+
+        // Update swarm listen addresses
+        if !self.swarm_host.is_empty() || self.swarm_port.is_some() {
+            let listen_array = doc["swarm"]["listen"].as_array_mut().unwrap();
+
+            for item in listen_array.iter_mut() {
+                let addr: Multiaddr = item.as_str().unwrap().parse().unwrap();
+                let mut new_addr = Multiaddr::empty();
+
+                for protocol in addr.iter() {
+                    match (&protocol, ipv4_host, ipv6_host, self.swarm_port) {
+                        (Protocol::Ip4(_), Some(ipv4_host), _, _) => {
+                            new_addr.push(Protocol::Ip4(ipv4_host));
+                        }
+                        (Protocol::Ip6(_), _, Some(ipv6_host), _) => {
+                            new_addr.push(Protocol::Ip6(ipv6_host));
+                        }
+                        (Protocol::Tcp(_) | Protocol::Udp(_), _, _, Some(new_port)) => {
+                            new_addr.push(match &protocol {
+                                Protocol::Tcp(_) => Protocol::Tcp(new_port),
+                                Protocol::Udp(_) => Protocol::Udp(new_port),
+                                _ => unreachable!(),
+                            });
+                        }
+                        _ => new_addr.push(protocol),
+                    }
+                }
+
+                *item = Value::from(new_addr.to_string());
+            }
+        }
+
+        // Update server listen addresses
+        if !self.server_host.is_empty() || self.server_port.is_some() {
+            let listen_array = doc["server"]["listen"].as_array_mut().unwrap();
```

Comment:
all these `unwrap` points should have proper error handling

Reply:
Sure, I think `unwrap_or_default()` doesn't really make sense, so I'm going to put `ok_or()`